File: Data Structures & Algorithms/k-closest-points-to-origin/submission-0.java
class Solution {
    public int[][] kClosest(int[][] points, int k) {
        PriorityQueue<int[]> minHeap = new PriorityQueue<>(Comparator.comparing(a -> a[0]));

        // add items to heap
        for(int i = 0; i < points.length; i++){
            int[] point = points[i];
            int dist = point[0] * point[0] + point[1] * point[1];
            minHeap.offer(new int[]{ dist, point[0], point[1]});
        }

        int[][] res = new int[k][2];
        for( int i = 0 ; i < k; i++){
            int[] point = minHeap.poll();
            res[i] = new int[]{ point[1], point[2]};
        }

        return res;
    }
}

File: Data Structures & Algorithms/palindrome-number/submission-0.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False
        
        sx = str(x)
        l, r = 0, len(sx) - 1
        while r > l:
            if sx[l] != sx[r]:
                return False
            l += 1
            r -= 1

        return True